### PR TITLE
CodeCompletion: Annotate archetypes, generic parameters and dependent…

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
… members

The original goal was to have IDE annotate these in code completion descriptions, but the fixes to ASTPrinter ended up impacting the annotated output of some other SourceKit requests, likely and hopefully for the good.

### CODE REVIEW

File: test/SourceKit/CursorInfo/cursor_info.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import Foo
+ import Foo
```
Comment (rintaro) [inline test/SourceKit/CursorInfo/cursor_info.swift:1]: Unrelated change

**Review by rintaro — APPROVED** [review]
This looks good to me. @nathawes Any concern?

**Review by nathawes — APPROVED** [review]
LGTM too - thanks for improving this!

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by AnthonyLatsis** [comment]
@swift-ci please test

**Comment by AnthonyLatsis** [comment]
@swift-ci please test
